test/stats.node.js

```diff
@@ -18,131 +19,188 @@ const tryEcho = utils.tryEcho
 const Switch = require('../src')
 
 describe('Stats', () => {
-  let switchA
-  let switchB
-
-  before((done) => createInfos(2, (err, infos) => {
-    expect(err).to.not.exist()
+  const setup = (cb) => {
+    createInfos(2, (err, infos) => {
+      expect(err).to.not.exist()
 
-    const options = {
-      stats: {
-        computeThrottleTimeout: 100
+      const options = {
+        stats: {
+          computeThrottleTimeout: 100
+        }
       }
-    }
-
-    const peerA = infos[0]
-    const peerB = infos[1]
-
-    peerA.multiaddrs.add('/ip4/127.0.0.1/tcp/9001')
-    peerB.multiaddrs.add('/ip4/127.0.0.1/tcp/9002')
-
-    switchA = new Switch(peerA, new PeerBook(), options)
-    switchB = new Switch(peerB, new PeerBook(), options)
-
-    switchA.transport.add('tcp', new TCP())
-    switchB.transport.add('tcp', new TCP())
-
-    switchA.connection.crypto(secio.tag, secio.encrypt)
-    switchB.connection.crypto(secio.tag, secio.encrypt)
 
-    switchA.connection.addStreamMuxer(multiplex)
-    switchB.connection.addStreamMuxer(multiplex)
+      const peerA = infos[0]
+      const peerB = infos[1]
+
+      peerA.multiaddrs.add('/ip4/127.0.0.1/tcp/9001')
```

Comment:
This is gonna conflict if we run two test cases at the same time, is it possible to set the port to `0` and then later get it (if you need it) instead? Think you're not actually using the port in the test, so should work fine to be set to `0.`

Reply:
@VictorBjelkholm good call, fixed!